README.md

```diff
@@ -1 +1,51 @@
 # purescript-halogen-menu
+A reusable halogen component that presents an interactive menu.
+
+## HTML
+Menus are rendered as unordered lists of submenus which can be selected and dismissed. Selected submenus are also rendered as unordered lists.
+
+Here is an example of a rendered menu with the first submenu selected.
+
+```HTML
+<ul><li><button>Color</button><ul><li><button>Load color</button></li><li><button>Save color</button></li></ul></li><li><div><button>Edit</button></div></li></ul>
+```
+
+## Guide
+
+### Installing `purescript-halogen-menu`
+* Install `purescript-markdown-halogen` using bower: `bower i purescript-markdown-halogen --save`.
+
+### Rendering a menu
+* Create a Halogen parent component to install your menu in.
+* Decide what you want selecting an item from your menu to do. Commonly this will be to cause some other component or to evaluate a query.
+* Define a query or other data type which represents what you want selecting an item from your menu to do.
+* Import `Halogen.Menu.Model (Menu(), makeMenu)`.
+* Construct the intitial state of your menu as in [this example](https://github.com/beckyconning/color-editor/tree/94de4b0297ef1ed6e76561ec88234be0ca7f07bd/src/ColorEditorMenu/Model.purs) replacing `ColorQuery Unit` with the type you defined in the previous step.
+* Import `Halogen.Menu.Component (MenuP(), MenuQueryP(), SubmenuSlotAddress(), menuComponent)`.
+* Define a slot address for your menu.
+* Define a type synonym for your menu's state and queries using the query or other data type you defined above as in [this example](https://github.com/beckyconning/color-editor/tree/94de4b0297ef1ed6e76561ec88234be0ca7f07bd/src/ColorEditor/Component.purs#L37-L38).
```

Comment:
I think thats a good idea but I'd like there to be gulp stuff to compile and serve the examples easily. This isn't simple to do with pulp so I split them up into two repositories. I kinda like the simplicity of having two separate repos... I'd be equally happy to raise integrating the examples into this repo as an issue but I'm not sure its the best use of time pre this pr.